src/pdmat.jl

```diff
@@ -2,23 +2,32 @@
 Full positive definite matrix together with a Cholesky factorization object.
 """
 struct PDMat{T<:Real,S<:AbstractMatrix} <: AbstractPDMat{T}
-    dim::Int
     mat::S
     chol::Cholesky{T,S}
 
-    PDMat{T,S}(d::Int,m::AbstractMatrix{T},c::Cholesky{T,S}) where {T,S} = new{T,S}(d,m,c)
+    PDMat{T,S}(m::AbstractMatrix{T},c::Cholesky{T,S}) where {T,S} = new{T,S}(m,c)
+    function PDMat{T,S}(d::Int, m::AbstractMatrix{T},c::Cholesky{T,S}) where {T,S}
```

Comment:
Can we deprecate this method?

Reply:
I tend to think of deprecations as breaking.

Reply:
IIRC historically in PDMats deprecations were considered non-breaking and removed in one of the following breaking releases.

Reply:
Okay, I've added deprecations.